Sources/Drain.swift

```diff
@@ -48,9 +48,16 @@ public final class Drain: DataRepresentable, Stream {
         return true
     }
 
-    public func receive() throws -> Data {
-        close()
-        return buffer
+    public func receive(maxBytes: Int) throws -> Data {
```

Comment:
Yeah I think making this optional would be nice actually. What do you think @paulo?